projects/signals.py

```diff
@@ -1,12 +1,58 @@
 from django.db.models.signals import post_save, post_delete
 from django.dispatch import receiver
-from .models import Project
-from .documents import ProjectDocument
+
+from projects.models import Project, ProjectHistory
+from projects.documents import ProjectDocument
+
+from investments.models import Subscription
+from communications.services import create_in_app_notification
+
+TRACKED_FIELDS = ['title', 'description', 'funding_goal', 'status', 'website', 'technologies_used', 'milestones']
 
 @receiver(post_save, sender=Project)
-def update_project(sender, instance, **kwargs):
-    ProjectDocument().update(instance)
+def handle_project_updates(sender, instance, created, **kwargs):
+    """
+    Tracks project updates, creates a history record, and sends
+    notifications to subscribed investors.
+    """
+    if created:
+        return
+
+    try:
+        old_instance = instance._pre_save_instance
```

Comment:
The signal handler does not handle the case when old_instance is None, which may cause errors. Add a guard clause to return early if old_instance is missing.
```
if not old_instance:
    return
```